### PR TITLE
[AppService] fix #11412: az webapp create is removing existing application settings

### LINKED ISSUE #11412: az webapp create is removing existing application settings
## Describe the bug
When `az webapp create` is called every time a new pipeline is running, all existing appsettings is removed and it's force the site to break down if ex. WEBSITE_RUN_FROM_PACKAGE=1

**Command Name**
`az webapp create`

**Errors:**
See above description.

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az webapp create --name {} --plan {}`

Trough the portal or anywhere you can add an application setting do that. Then run the same command again.

- `az webapp create --name {} --plan {}`

Now the application setting you created before is deleted.

We are using Azure CLI in our CI/CD pipeline so we are running the same command over and over again. Previous we used Azure ARM, which has not the same issue.

## Expected Behavior
Keep existing application settings.

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Shell: powershell.exe

azure-cli 2.0.76 *

Extensions:
azure-devops 0.8.0
dev-spaces-preview 0.1.6
interactive 0.4.3

```
## Additional Context

It looks like `az webapp deployment slot create` is not removing existing application settings when called, so it would be cool if `az webapp create` can act the same way.

### PR DESCRIPTION
**Pull request by Kotasudhakarreddy**
FIXES #11412 
**Description of PR (Mandatory)**  
(Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process)

**Testing Guide**  
(Example commands with explanations)

**History Notes:**  
(Fill in the following template if multiple notes are needed, otherwise PR title will be used for history note.)

[Component Name 1] (BREAKING CHANGE:) (az command:) make some customer-facing change.  
[Component Name 2] (BREAKING CHANGE:) (az command:) make some customer-facing change.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by Kotasudhakarreddy** [comment]
@yungezz - #12730 has been Closed as incomplete. Please review this.

@panchagnula - FYI

**Comment by Kotasudhakarreddy** [comment]
> LGTM now, could you pls add tests?

@yungezz - here the build is failing with errors. I am not sure on it. if you can,could you please help me to understand what is wrong with my PR.

**Comment by yungezz** [comment]
> > LGTM now, could you pls add tests?
> 
> @yungezz - here the build is failing with errors. I am not sure on it. if you can,could you please help me to understand what is wrong with my PR.

hi @qwordy could you pls help on this? thanks.

**Comment by qwordy** [comment]
> 
> 
> > LGTM now, could you pls add tests?
> 
> @yungezz - here the build is failing with errors. I am not sure on it. if you can,could you please help me to understand what is wrong with my PR.

1. Pull dev branch to your branch.
2. Can you pass locally?

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -150,7 +150,19 @@ def create_webapp(cmd, resource_group_name, name, plan, runtime=None, startup_fi
     if language is not None and language.lower() == 'dotnetcore':
         site_config.app_settings.append(NameValuePair(name='ANCM_ADDITIONAL_ERROR_PAGE_LINK',
                                                       value='https://{}.scm.azurewebsites.net/detectors'.format(name)))
-
+    # This is to keep the existing appsettings for a newly created webapp on existing webapp name.
+    name_validation = client.check_name_availability(name, 'Site')
+    if not name_validation.name_available:
+        existing_app_settings = _generic_site_operation(cmd.cli_ctx, resource_group_name,
+                                                        name, 'list_application_settings')
+        # To fetch the newly added appsetting names
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: This seems be rather complex and non performant way to fix this issue. The problem here is this line https://github.com/Azure/azure-cli/blob/7b849de40f3958a2a69b7b5ef63a5968bb3d8bf6/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L94
   ` site_config = SiteConfig(app_settings=[])`
when ever the create_webapp us called for an existing app we always reset the app settings to [] , i would fix this by making a call to get list of appsettings (that you are already doing here) if the call returns 404 exception - then i would set this value to [] as is, if not set this to value returned from the API - & leave the rest as is.

**Comment by panchagnula** [comment]
@Kotasudhakarreddy any reason you closed this PR? Don't we need this to be merged?

**Comment by qwordy** [comment]
> 
> 
> @Kotasudhakarreddy any reason you closed this PR? Don't we need this to be merged?

He has a incorrect commit history in this PR. And the changes are hard to review because they contain so many changes that are not related to this PR.

**Comment by panchagnula** [comment]
> > @Kotasudhakarreddy any reason you closed this PR? Don't we need this to be merged?
> 
> He has a incorrect commit history in this PR. And the changes are hard to review because they contain so many changes that are not related to this PR.

Here is the new PR that was created https://github.com/Azure/azure-cli/pull/12865 - please review this. Thank you.

**Pull request closed without merging**